src/backend/distributed/master/master_expire_table_cache.c

```diff
@@ -174,16 +156,22 @@ DropShardsFromWorker(WorkerNode *workerNode, Oid relationId, List *shardInterval
 
 		AppendShardIdToName(&shardName, shardInterval->shardId);
 		quotedShardName = quote_qualified_identifier(schemaName, shardName);
-		appendStringInfo(workerCommand, "%s", quotedShardName);
 
-		/* append a comma after the shard name if there are more shards */
-		if (lnext(shardIntervalCell) != NULL)
+		if (relationKind == RELKIND_RELATION)
 		{
-			appendStringInfo(workerCommand, ", ");
+			appendStringInfo(workerCommand, DROP_REGULAR_TABLE_COMMAND, quotedShardName);
+		}
+		else if (relationKind == RELKIND_FOREIGN_TABLE)
+		{
+			appendStringInfo(workerCommand, DROP_FOREIGN_TABLE_COMMAND, quotedShardName);
+		}
+		else
+		{
+			ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
+							errmsg("expire target is not a regular or foreign table")));
 		}
-	}
-
-	appendStringInfo(workerCommand, " CASCADE");
 
-	ExecuteRemoteCommand(workerNode->workerName, workerNode->workerPort, workerCommand);
+		ExecuteRemoteCommand(workerNode->workerName, workerNode->workerPort,
```

Comment:
Hmz, dropping all shards in one go is much more efficient than dropping them one by one and although it doesn't matter much here, it's what we would want to do in other cases as well.  What if we append cascade in the other places that use DROP_REGULAR_TABLE_COMMAND and leave this function as is?